include/openenclave/internal/raise.h

```diff
@@ -85,20 +93,64 @@ OE_EXTERNC_BEGIN
 // variadic paramter is empty in OE_RAISE_MSG(RESULT, fmt, ...)
 // eg : OE_RAISE_MSG(OE_FAILURE, "your message", NULL);
 
-#define OE_RAISE_MSG(RESULT, fmt, ...)   \
-    do                                   \
-    {                                    \
-        result = (RESULT);               \
-        if (result != OE_OK)             \
-        {                                \
-            OE_TRACE_ERROR(              \
-                fmt " (oe_result_t=%s)", \
-                ##__VA_ARGS__,           \
-                oe_result_str(result));  \
-        }                                \
-        goto done;                       \
+/* Note: on Linux, the above example fails, as the NULL pointer is fed as the
+ * value for oe_result_t in log, showing invalid (null) output.
+ *
+ * A solution is to define OS specfic versions of OE_RAISE_MSG() that
+ * handle cases with and without NULL, to work properly on all OSes while
+ * avoiding the need to modify existing code.
+ */
+
+#if defined(_MSC_VER)
```

Comment:
Line 104 doesn't quite match what line 99-101 says.  _MSC_VER is a compiler-specific, not OS-specific, preprocessor symbol.  Either update 99-101 to talk about compiler variations (if it's a compiler issue), or (if it's really an OS issue) 104-105 should instead test the platform like if defined(_WIN64).

Reply:
Good point, this is compiler issue. Will fix.

Reply:
Great, make sure to update line 129 too then.

Reply:
That has been updated too.